xbmc/dialogs/GUIDialogMediaSource.cpp

```diff
@@ -473,7 +473,8 @@ void CGUIDialogMediaSource::UpdateButtons()
     return;
 
   CONTROL_ENABLE_ON_CONDITION(CONTROL_OK, !m_paths->Get(0)->GetPath().empty() && !m_name.empty());
-  CONTROL_ENABLE_ON_CONDITION(CONTROL_PATH_ADD, !m_paths->Get(0)->GetPath().empty());
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_PATH_ADD, !m_paths->Get(0)->GetPath().empty() &&
+                                                    !(!m_paths->IsEmpty() && m_type == "files"));
```

Comment:
simplified:
```suggestion
  CONTROL_ENABLE_ON_CONDITION(CONTROL_PATH_ADD,
                              m_type != "files" && !m_paths->Get(0)->GetPath().empty());
```

!m_paths->IsEmpty() is not correct anyway, see condition at the top of the function
There is always at least one item in the list so m_paths->IsEmpty() never returns true.

Defining an enum for the type values would be worthy but in a separate refactor PR if you feel like it.